internal/rpc/server/library/server.go

```diff
@@ -0,0 +1,42 @@
+package library
+
+import (
+	"context"
+
+	detector "github.com/aquasecurity/trivy/pkg/detector/library"
+
+	"github.com/google/wire"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy/internal/rpc"
+	"github.com/aquasecurity/trivy/pkg/log"
+	"github.com/aquasecurity/trivy/pkg/vulnerability"
+	proto "github.com/aquasecurity/trivy/rpc/detector"
+)
+
+var SuperSet = wire.NewSet(
+	detector.SuperSet,
+	vulnerability.SuperSet,
+	NewServer,
+)
+
+type Server struct {
+	detector   detector.Operation
+	vulnClient vulnerability.Operation
+}
+
+func NewServer(detector detector.Operation, vulnClient vulnerability.Operation) *Server {
+	return &Server{detector: detector, vulnClient: vulnClient}
+}
+
+func (s *Server) Detect(ctx context.Context, req *proto.LibDetectRequest) (res *proto.DetectResponse, err error) {
+	vulns, err := s.detector.Detect(req.FilePath, rpc.ConvertFromRpcLibraries(req.Libraries))
+	if err != nil {
+		log.Logger.Warn(err)
```

Comment:
I'd rather log or return error. Having both we might see duplicate error messages in the error log

Reply:
This is a function in a server side. So, if a server returns an error, the error will be displayed only in a client side. The error will not be duplicated. Or, do you mean the server should not display an error?

Reply:
My concern was that it's logged twice on the server side. Here and where `Detect()` with error handling.

Reply:
I see. `Detect()` is a method of an HTTP server.
https://github.com/aquasecurity/trivy/pull/303/files#diff-09f894f3b8925177d261b36dc04e31b9R56

 This error will be handled on a client side. It will not be logged twice on the server side.

Reply:
Got it!